### ISSUE TITLE
warning caused by test code

### ISSUE THREAD

**Issue by edwardhartnett** [NOAA-EMC/NCEPLIBS-bufr#335]
We need the test code to also compile without warning so we can turn on -Werror.

```
[ 90%] Building Fortran object test/CMakeFiles/outtest8_8.dir/outtest8.F90.o
cd /home/ed/NCEPLIBS-bufr/b/test && /usr/bin/gfortran -DKIND_8 -DUNDERSCORE -I/home/ed/NCEPLIBS-bufr/src -I/home/ed/NCEPLIBS-bufr/b/src/include/bufr_4 -g -fbacktrace -ggdb -Wall -O0 -fPIE -fdefault-real-8 -fdefault-integer-8 -c /home/ed/NCEPLIBS-bufr/test/outtest8.F90 -o CMakeFiles/outtest8_8.dir/outtest8.F90.o
/home/ed/NCEPLIBS-bufr/test/outtest8.F90:65:9:

   65 |   nbyt = lenbmg * 8
      |         1
Warning: Possible change of value in conversion from INTEGER(8) to INTEGER(4) at (1) [-Wconversion]

```

**Comment by jbathegit** [comment]
So here's an interesting little tidbit.  -Wall is already included as an option flag whenever CMAKE_BUILD_TYPE is specified as "Debug", and which in turn is how the build is already being done in the developer.yml workflow!  So, and unless I'm missing something, we're apparently already running with -Wall for that workflow.  So to clarify, is the idea here (and in #300) to turn on -Wall for **_all_** of the workflows (e.g. Intel, Linux, MacOS too)? Or was the intent just to do this for the gnu compiler in the developer workflow, and in which case it looks like this is already done!?

What's even more puzzling to me is that I was unable to reproduce the error you're showing above, at least running the following CMake flags on WCOSS2 (same as what we use in developer.yml) using the latest GNU gcc/12.1.0 package, and just manually adding in -Wall to see if it made any difference.  In other words, I used `-DCMAKE_Fortran_FLAGS="-g -fprofile-arcs -fprofile-abs-path -O0 -fsanitize=address -Wall" `and `-DCMAKE_C_FLAGS="-g -fprofile-arcs -fprofile-abs-path -O0 -fsanitize=address -Wall"` and `-DCMAKE_C_FLAGS=Debug`, and I didn't see a single warning pop up anywhere, including for outtest8_8 or any of the other test codes!?  Am I missing something here?

**Comment by edwardhartnett** [comment]
We will do this first for the GNU compiler in the developer workflow. The goal is to use -Wall -Werror, which will cause the build to error out on any warnings. Then, just as we have done with documentation, we will be able to prevent any new warnings from being committed to the code base.

Stand by and I'll take another look at this warning...

**Comment by jbathegit** [comment]
I understand the goal is to use -Wall -Werror, and that we have to clear all the -Wall alerts first before we can turn on -Werror, but the developer.yml workflow already uses the `-DCMAKE_C_FLAGS=Debug` option to cmake, which in turn already turns on -Wall for that build!  See the GNU `CMAKE_Fortran_FLAGS_DEBUG` and `CMAKE_C_FLAGS_DEBUG` options in CMakeLists.txt for more details.  So what is there still to do here?

Again, what's even more puzzling to me is that I couldn't reproduce the error you were seeing.  I looked at the text of your original post, and it looks like I was running with a lot more flags than you were, though I did notice you had a `-fPIE` flag set which I didn't have in my build.  Maybe that explains it, although I couldn't find any documentation for that flag anywhere so I'm not even sure what it's supposed to do(?)
